src/alchemlyb/estimators/bar_.py

```diff
@@ -150,13 +153,11 @@ def fit(self, u_nk):
             ad_delta += np.diagflat(np.array(dout), k=j + 1)
 
         # yield standard delta_f_ free energies between each state
-        self._delta_f_ = pd.DataFrame(
-            adelta - adelta.T, columns=self._states_, index=self._states_
-        )
+        self._delta_f_ = pd.DataFrame(adelta - adelta.T, columns=states, index=states)
 
         # yield standard deviation d_delta_f_ between each state
         self._d_delta_f_ = pd.DataFrame(
-            np.sqrt(ad_delta + ad_delta.T), columns=self._states_, index=self._states_
+            np.sqrt(ad_delta + ad_delta.T), columns=states, index=states
```

Comment:
I think at this point, we shall overwrite the default behaviour of BAR.states to the new values.

Reply:
I don't think so necessarily. self._states_ defines the number of lambda states in the columns, while states is capturing the number of states (for which the system is equilibrated to), so the number of unique fep-lambda values. So I guess the existing df._states_ this information is captures in df.columns, but I tried replacing it and there's a failure. I'm not willing to go through the code to reproduce that right now, but I think this is a larger question than should be answered in this PR.